### PR TITLE
[PWD-2529] Finalize Register Program Dialog

### PR DESCRIPTION
**Pull request by gangsararyop**
### What type of PR is this? (_check all applicable_)

- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update
- [ ] Setup
- [ ] Code Refactor
- [ ] Performance Improvements
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Revert

### Screenshots (_optional_)

network slow 3g biar keliatan loadingnya

https://github.com/PurwadhikaDev/purwadhika-corporate-website/assets/86463063/9453bd84-9d27-42fc-a8a4-2ef1798b0607

### CODE REVIEW

File: src/components/RegisterProgram/components/Content/DetailOrder.tsx [hunk 1]
```diff
@@ -102,42 +130,65 @@ const Order: FC<DataItem> = ({ qty = 0, totalPrice, data }) => {
...
-const Voucher: FC<DataItem> = ({ totalPrice }) => {
+const Voucher: FC<VoucherProps> = ({ loading, voucherBalance, totalPrice }) => {
+  const remainingVoucher = useMemo(() => {
+    const priceValue = totalPrice || 0
+
+    if (voucherBalance - priceValue < 0) {
+      return priceValue - voucherBalance
+    } else {
+      return voucherBalance - priceValue
+    }
```
Comment (katherinedavenia) [inline src/components/RegisterProgram/components/Content/DetailOrder.tsx:141]:
```suggestion
     return Math.abs(voucherBalance - priceValue);
```
bisa gini nihh setau aku, cobaindeh siapatau bisa

**Review by katherinedavenia — DISMISSED** [review]

File: src/components/RegisterProgram/index.tsx [hunk 2]
```diff
@@ -24,18 +28,21 @@ const RegisterProgram: FC<RegisterProgramProps> = ({
...
   const [activeStep, setActiveStep] = useState<number>(0)
 
-  const formik = useFormikRegister({ onClose, activeStep, setActiveStep })
+  const { data: voucher, isLoading: loadingVoucher } = useGetVoucherQuery()
+  const { data: program, isLoading: loadingProgram } = useGetProgram({
+    programId
+  })
+  const {
+    mutateAsync: corporateTokenTransaction,
+    isLoading: loadingTokenTransaction
```
Comment (kuntiarso) [inline src/components/RegisterProgram/index.tsx:39]: karna ini mutate function, tambahin prefix `create` / `update` di namingnya

```suggestion
    mutateAsync: createCorporateTokenTransaction,
    isLoading: loadingCreateTokenTransaction
```

File: src/features/token/types/CorporateTokenTransaction.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  attendeeEmail: string
+  attendeePhone: string
+}
+
+export interface CreateCorporateTokenTransactionProps {
+  programId: number
+  attendees: BodyAttendee[]
+}
+
+export interface CorporateTokenTransactionResponse {}
```
Comment (kuntiarso) [inline src/features/token/types/CorporateTokenTransaction.ts:19]: gak perlu dibuat kalo kosong

File: src/components/RegisterProgram/components/Content/DetailOrder.tsx [hunk 4]
```diff
@@ -58,21 +71,36 @@ const InfoText: FC<InfoTextProps> = ({
           valueWeight ? `${valueWeight} !important` : '400 !important'
         }
       >
-        {minus && '-'}Rp{value.toLocaleString('id')}
+        {loading ? (
+          <Skeleton variant="text" component="span" width={100} />
+        ) : (
+          `${isMinus}Rp${value.toLocaleString('id')}`
```
Comment (kuntiarso) [inline src/components/RegisterProgram/components/Content/DetailOrder.tsx:77]:
```suggestion
          `${isMinus}${toRupiah(value)}`
```

File: src/components/RegisterProgram/components/Content/DetailOrder.tsx [hunk 5]
```diff
@@ -58,21 +71,36 @@ const InfoText: FC<InfoTextProps> = ({
...
+            title
+          )}
+        </Typography>
+
         <Typography variant="captionregular" color="neutral400.main">
-          {qty} x Rp{price.toLocaleString('id')}
+          {loading ? (
+            <Skeleton variant="text" component="span" width={120} />
+          ) : (
+            `${qty} x Rp${price.toLocaleString('id')}`
```
Comment (kuntiarso) [inline src/components/RegisterProgram/components/Content/DetailOrder.tsx:102]:
```suggestion
            `${qty} x ${toRupiah(price)}`
```

**Review by kuntiarso — COMMENTED** [review]
sorry kelewat wkwk pake toRupiah aja udah ada di `utils`

File: src/components/RegisterProgram/components/ModalAction.tsx [hunk 6]
```diff
@@ -1,26 +1,49 @@
...
+    return totalPayment < 0
+  }, [totalPrice, voucher])
+
   return (
     <DialogActions sx={{ p: 3 }}>
       {activeStep === 0 ? (
         <Button onClick={() => handleSubmit()}>Lanjut</Button>
       ) : (
-        <Button onClick={() => handleSubmit()}>Daftar Sekarang</Button>
+        <LoadingButton
```
Comment (FahemHakikiKhaya) [inline src/components/RegisterProgram/components/ModalAction.tsx:39]: size button

Comment (gangsararyop) [reply src/components/RegisterProgram/components/ModalAction.tsx:39]: pake size default hem itu, ga perlu dideclare size nya

**Review by kuntiarso — DISMISSED** [review]
🔆🔆

**Review by kuntiarso — APPROVED** [review]

**Review by Vesals — DISMISSED** [review]

**Review by Vesals — APPROVED** [review]

**Review by haikal1223 — APPROVED** [review]

**Pull request merged**